### ISSUE TITLE
the radian in vscode could not recognize the directory that I set in ‘Rprofile.site’

### ISSUE THREAD

**Issue by mewinners** [REditorSupport/vscode-R#1122]
Most of time, the radian only in vscode could not recognize the directory that I set in ‘Rprofile.site’. It recognized successfully only once, but I don’t know why, because I didn’t change any settings. Here are some relating screenshots. I am looking forward to your reply.

Radian in vscode:
![1](https://user-images.githubusercontent.com/74438905/172033887-d74be394-dad9-4809-aaf6-7a96f6ff14a5.png)

Radian in windows terminal (powershell and cmd):
![2](https://user-images.githubusercontent.com/74438905/172033899-125c395f-d04f-4334-b032-e190dd31c1b4.png)
![3](https://user-images.githubusercontent.com/74438905/172033901-479c1138-b988-48e6-bab4-6a7412e92f91.png)

R studio:
![4](https://user-images.githubusercontent.com/74438905/172033903-aeab9a0c-9408-440a-8b65-f78944393d2e.png)

Rprofile.site:
![5](https://user-images.githubusercontent.com/74438905/172033911-46619c0f-87d5-4bd6-9a5a-9c373d2a205a.png)

Settings relating radian in vscode:
![6](https://user-images.githubusercontent.com/74438905/172033915-8bd3b44f-ad1a-4430-a7ce-45e5469f0458.png)
![7](https://user-images.githubusercontent.com/74438905/172033916-62c23e5a-b873-4e21-94e7-3769b9586153.png)
![8](https://user-images.githubusercontent.com/74438905/172033917-4d502691-b21b-4203-a3fb-8f3afb55cad9.png)

**Comment by renkun-ken** [comment]
Does it occur with official R console in vscode?

**Comment by renkun-ken** [comment]
Is your `Rprofile.site` located at the site directory? Since your `r.rterm.options` specifies `--no-site-file`, it will not be sourced on R session startup.

**Comment by renkun-ken** [comment]
Is there a reason not to use `.Rprofile` in your case?

**Comment by mewinners** [comment]
> Does it occur with official R console in vscode?

yes.

**Comment by mewinners** [comment]
> Is your `Rprofile.site` located at the site directory? Since your `r.rterm.options` specifies `--no-site-file`, it will not be sourced on R session startup.

It is located at R\R-4.2.0\etc\Rprofile.site, indeed. So,how can I solve this problem?

**Comment by mewinners** [comment]
> Is there a reason not to use `.Rprofile` in your case?

Probably no. I am just used to change the setting in `Rprofile.site`

**Comment by renkun-ken** [comment]
Does it work if you use `~/.Rprofile` to specify the same settings?

**Comment by mewinners** [comment]
> 

I will have a try, But I am not familiar with `.Rprofile`, is there any way use `Rprofile.site`?

**Comment by renkun-ken** [comment]
You should restore the default value of `r.rterm.option` in your vscode settings.

**Comment by mewinners** [comment]
> You should restore the default value of `r.rterm.option` in your vscode settings. Looks like it is `--no-init-file` prevents R session from sourcing `Rprofile.site` on startup.

I have changed the `.Rprofile` ,radian in vscode successfully recognizes the directory in R. Thank you very much!
And,
I followed the user guide of Radian, changed the `r.rterm.option`, removed `--no-save `and `--no-store`, then add '--no-site-file'. In your opinion, I shoud restore the `r.rterm.option`, re-add `--no-save `and `--no-store` ? or just remove the '--no-site-file', change `r.rterm.option` to blank?

**Comment by mewinners** [comment]
> Does it work if you use `~/.Rprofile` to specify the same settings?
> 
> As you specifiy `--no-init-file` in your vscode setting `r.rterm.option`, your site file will not be removed. You should use the default value of that that setting unless you intend so.

I have changed the .Rprofile ,radian in vscode successfully recognizes the directory in R. Thank you very much!
Here I have another problem, I run the rscript in windows ternimal with radian could display the plot I need, but there is no plot displaying while I run the same rscript in vscode with radian.

**Comment by renkun-ken** [comment]
We recommend using [httpgd](https://nx10.github.io/httpgd/) to show plot. You might want to install `httpgd` and turn on `r.useHttpgd` in your vscode settings.

**Comment by mewinners** [comment]
> We recommend using [httpgd](https://nx10.github.io/httpgd/) to show plot. You might want to install `httpgd` and turn on `r.useHttpgd` in your vscode settings.

I have already done those.

**Comment by renkun-ken** [comment]
Does session watcher attach to your R session successfully? You could run `ls.str(.vsc)` in your R terminal.

**Comment by mewinners** [comment]
emmm,
`Error in ls.str(.vsc) : object '.vsc' not found`

**Comment by renkun-ken** [comment]
Are you using `R: Create R terminal` to start a new R terminal?

**Comment by mewinners** [comment]
> Are you using `R: Create R terminal` to start a new R terminal?

no, just radian

**Comment by renkun-ken** [comment]
Then you should follow the steps at <https://github.com/REditorSupport/vscode-R/wiki/R-Session-watcher#advanced-usage-for-self-managed-r-sessions> to make session watcher work with self-managed R terminals.

**Comment by mewinners** [comment]
> Are you using `R: Create R terminal` to start a new R terminal?

I have set the PATH environment variable of anaconda, install the radian through conda.

**Comment by mewinners** [comment]
> Then you should follow the steps at https://github.com/REditorSupport/vscode-R/wiki/R-Session-watcher#advanced-usage-for-self-managed-r-sessions to make session watcher work with self-managed R terminals.

OK, I will have a try, thank you.

**Comment by mewinners** [comment]
> Then you should follow the steps at https://github.com/REditorSupport/vscode-R/wiki/R-Session-watcher#advanced-usage-for-self-managed-r-sessions to make session watcher work with self-managed R terminals.

It didn't work.
I have appended the following code to the `.Rprofile`:
if (interactive() && Sys.getenv("RSTUDIO") == "") {
  source(file.path(Sys.getenv(if (.Platform$OS.type == "windows") "USERPROFILE" else "HOME"), ".vscode-R", "init.R"))
}

**Comment by renkun-ken** [comment]
Would you like to see the value of

```r
file.path(Sys.getenv(if (.Platform$OS.type == "windows") "USERPROFILE" else "HOME"), ".vscode-R", "init.R")
```

and see if the file exists?

If so, what if you manually source the file in a new session?

**Comment by mewinners** [comment]
> Would you like to see the value of
> 
> ```r
> file.path(Sys.getenv(if (.Platform$OS.type == "windows") "USERPROFILE" else "HOME"), ".vscode-R", "init.R")
> ```
> 
> and see if the file exists?
> 
> If so, what if you manually source the file in a new session?

Yes, it locates 'D:\Program Files\Microsoft Visual Studio Code\data\extensions\reditorsupport.r-2.5.0\R\session\init.R'
but I don't know how to manually source the file?

**Comment by renkun-ken** [comment]
Then you could try starting a new R session, and run

```r
source("D:\\Program Files\\Microsoft Visual Studio Code\\data\\extensions\\reditorsupport.r-2.5.0\\R\\session\\init.R")
```

**Comment by mewinners** [comment]
> Then you could try starting a new R session, and run
> 
> ```r
> source("D:\\Program Files\\Microsoft Visual Studio Code\\data\\extensions\\reditorsupport.r-2.5.0\\R\\session\\init.R")
> ```

still not work.
but I find that, if I start a new terminal in vscode, and copy the rscript in it and run, there will be a window to display the plot.
![1](https://user-images.githubusercontent.com/74438905/172042373-f2c6b826-0c97-4cfd-8e30-107ecfe17309.png)

if I use ctrl+enter in the r file in vscode, there will no windows

**Comment by renkun-ken** [comment]
It is strange that your `USERPROFILE` does not point to a user folder but the installation location of the extension.

**Comment by mewinners** [comment]
> It is strange that your `USERPROFILE` does not point to a user folder but the installation location of the extension.

maybe there are some errors in the path.
By the way, does this picture come from httpgd? mean that the httpgd was successfully applied?
![](https://user-images.githubusercontent.com/74438905/172042373-f2c6b826-0c97-4cfd-8e30-107ecfe17309.png)

**Comment by renkun-ken** [comment]
It is R's built-in plot window, not from vscode-R or httpgd.

**Comment by mewinners** [comment]
> It is R's built-in plot window, not from vscode-R or httpgd.

emmm, so it failed again.
and I found this, does it have influences?
![Snipaste_2022-06-05_22-34-03](https://user-images.githubusercontent.com/74438905/172055764-8975266c-c318-4234-916d-af98c738322c.png)
